src/proc_diskstats.c

```diff
@@ -523,6 +678,61 @@ int do_proc_diskstats(int update_every, unsigned long long dt) {
 		}
 
 		// --------------------------------------------------------------------
+
+		if(ddo_space) {
+			if(mount_point) {
+				if (statvfs(family, &buff_statvfs) < 0) {
+					error("Faild checking disk space usage of %s", family);
+				} else {
+					space_avail = buff_statvfs.f_bavail * buff_statvfs.f_bsize;
+					space_avail_root = (buff_statvfs.f_bfree - buff_statvfs.f_bavail) * buff_statvfs.f_bsize;
+					space_used = (buff_statvfs.f_blocks - buff_statvfs.f_bfree) * buff_statvfs.f_bsize;
+				}
+
+				st = rrdset_find_bytype("disk_space", disk);
```

Comment:
I think there are exotic filesystems out there. Even for btrfs getting the disk free space is not that simple. I guess a few filesystems will return just zero because "there cannot be a value for disk free space".

I guess this is what the ONDEMAND should do. ONDEMAND was just an idea to prevent adding charts that are always zero.